nengo/networks/workingmemory.py

```diff
@@ -5,15 +5,13 @@
 
 
 def InputGatedMemory(n_neurons, dimensions, fdbk_scale=1.0, gate_gain=10,
-                     difference_gain=1.0, reset_gain=3,
-                     mem_config=None, net=None):
+                     difference_gain=1.0, reset_gain=3, net=None):
     """Stores a given vector in memory, with input controlled by a gate."""
     if net is None:
         net = nengo.Network(label="Input Gated Memory")
 
-    if mem_config is None:
-        mem_config = nengo.Config(nengo.Connection)
-        mem_config[nengo.Connection].synapse = nengo.Lowpass(0.1)
+    mem_config = nengo.Config(nengo.Connection)
+    mem_config[nengo.Connection].synapse = nengo.Lowpass(0.1)
```

Comment:
I'd vote to keep it out of the parameter list, mainly because the
constructor is already pretty bulky. In fact I'd probably take out
reset_gain and gate_gain as well, because those are similarly things that
only a really specific user would want to adjust (I do it very rarely, and
I use input gated memories a lot). But I won't push hard for that.
On 19 Aug 2015 23:03, "tcstewar" <notifications@github.com> wrote:

> In nengo/networks/workingmemory.py
> <https://github.com/nengo/nengo/pull/814#discussion_r37491343>:
>
> >      """Stores a given vector in memory, with input controlled by a gate."""
> >      if net is None:
> >          net = nengo.Network(label="Input Gated Memory")
> >
> > -    if mem_config is None:
> > -        mem_config = nengo.Config(nengo.Connection)
> > -        mem_config[nengo.Connection].synapse = nengo.Lowpass(0.1)
> > +    mem_config = nengo.Config(nengo.Connection)
> > +    mem_config[nengo.Connection].synapse = nengo.Lowpass(0.1)
>
> Blarg. I'd take an additional parameter over an extra for-loop any day.
>
> It all depends on how often such an additional parameter would be used. :)
>
> But yeah, I think this is a borderline case. But we really should remember
> that we don't need parameters for *everything*, and even something like
> this isn't that hard to go find in the object model. So I think it's more
> of a note that hey, don't worry too much about putting in parameters for
> everything -- just put in parameters for things that you need to. After
> all, we can always add parameters in the future (in a backwards-compatible
> way) if it turns out we use it a lot.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/nengo/nengo/pull/814/files#r37491343>.
>